django_short_urls/tests/auth_test.py

```diff
@@ -0,0 +1,50 @@
+# coding=utf-8
+
+from __future__ import unicode_literals
+
+from django_app.test import PyW4CTestCase
+from mock import Mock
+
+from django_short_urls.auth import _safe_base64_decode, login_with_basic_auth_required
+from django_short_urls.models import User
+from http.status import HTTP_UNAUTHORIZED
+
+
+@login_with_basic_auth_required
+def _test_view(request):  # pylint: disable=unused-argument
+    """Test view that does nothing"""
+    raise NotImplementedError("You can't be here")
+
+
+class AuthTestCase(PyW4CTestCase):
+    def test__safe_base64_decode(self):
+        # Basic test
+        self.assertEqual(_safe_base64_decode('bG9naW46cGFzc3dvcmQ='), 'login:password')
+        # Add some unicode in there
+        self.assertEqual(_safe_base64_decode('w6lsw6lwaGFudA=='), 'éléphant')
+        # Insufficient padding, recover gracefully
+        self.assertEqual(_safe_base64_decode('w6lsw6lwaGFudA'), 'éléphant')
+        # Try decoding something that's not base64 encoded, return an empty string
+        self.assertEqual(_safe_base64_decode('login:password'), '')
+        # Invalid utf8, raise an error
+        with self.assertRaises(UnicodeDecodeError):
+            print _safe_base64_decode('77+97+9')
```

Comment:
The function expects the auth credentials to be encoded as `utf8`. This, decoded, is invalid `utf8`, which in turn leads to the function raising a `UnicodeDecodeError`. This ensures that we don't capture all exceptions in the function.